### PR TITLE
Generate cpp docs

### PR DESCRIPTION
**Pull request by svekars**

### CODE REVIEW

File: fbgemm_gpu/README.md [hunk 1]
```diff
@@ -156,6 +156,87 @@ parts of FBGEMM_GPU please see our Wiki (work in progress).
 We have extensively used comments in our source files. The best and up-do-date
 documentation is available in the source files.
 
+# Building API Documentation
```
Comment (jianyuh) [inline fbgemm_gpu/README.md:159]: We can put this as a separate README.md in fbgemm_gpu/docs folder, and then link to that file from this README.md (add a ## subsection on docs and link to there)?

File: .github/workflows/fbgemm_docs.yml [hunk 2]
```diff
@@ -63,6 +63,10 @@ jobs:
       run: |
         conda run -n build_binary \
           python -c "import fbgemm_gpu"
+    - name: Install Doxygen
+      run: |
+        sudo apt-get install doxygen
```
Comment (geyyer) [inline .github/workflows/fbgemm_docs.yml:68]: Do we need doxygen on the system level or "conda install -n build_binary -c conda-forge doxygen" would be enough?

Comment (svekars) [reply .github/workflows/fbgemm_docs.yml:68]: On the env level should be enough!

File: fbgemm_gpu/docs/source/cpp-api/merge_pooled_embeddings.rst [hunk 3]
```diff
@@ -0,0 +1,20 @@
+Pooled Embeddings Operators
```
Comment (geyyer) [inline fbgemm_gpu/docs/source/cpp-api/merge_pooled_embeddings.rst:1]: I'm not sure what goes wrong, but the page builds with warnings https://deploy-preview-1165--eclectic-stroopwafel-199537.netlify.app/cpp-api/merge_pooled_embeddings.html

**Review by geyyer — COMMENTED** [review]
@svekars thank you, the PR looks good to me! Could you address the @jianyuh 's comment and let's make sure the docs are complete before publishing!

File: fbgemm_gpu/README.md (hunk 1 above)
Comment (svekars) [reply fbgemm_gpu/README.md:159]: moved to docs/README.md

File: fbgemm_gpu/docs/source/cpp-api/merge_pooled_embeddings.rst (hunk 3 above)
Comment (svekars) [reply fbgemm_gpu/docs/source/cpp-api/merge_pooled_embeddings.rst:1]: https://deploy-preview-1165--eclectic-stroopwafel-199537.netlify.app/cpp-api/merge_pooled_embeddings.html fixed

**Review by jianyuh — APPROVED** [review]
Thanks @svekars for the awesome supports on documentations for FBGEMM! Let's start importing the PR into an internal Diff and start the landing process.

**Comment by jianyuh** [comment]
@svekars I think it's good to keep undocumented functions in HTML and we can follow up on adding more descriptions for these APIs. BTW, if we want to hide them, should we just add `#ifndef DOXYGEN_THIS_WILL_BE_SKIPPED`?

**Comment by svekars** [comment]
> BTW, if we want to hide them, should we just add `#ifndef DOXYGEN_THIS_WILL_BE_SKIPPED`?
Yes, you add this to the source files:
```
#ifndef DOXYGEN_THIS_WILL_BE_SKIPPED
code/documentation that you want to skip
#endif
```
All source files are listed under `INPUT` in the `fbgemm_gpu/docs/Doxygen.in` file.

**Pull request closed without merging**